lib/src/types/binary_codec.dart

```diff
@@ -505,6 +628,35 @@ class PostgresBinaryEncoder {
     }
     return writer.toBytes();
   }
+
+  Uint8List _encodeRange<T>(
+      Range<T> range, Encoding encoding, int elementTypeOid) {
+    final buffer = BytesBuffer();
+    buffer.add([range.flag]);
+    switch ((range.lower == null, range.upper == null)) {
+      case (false, false):
+        if (range.flag == 1) break;
+        _encodeRangeValue(range.lower, encoding, elementTypeOid, buffer);
+        _encodeRangeValue(range.upper, encoding, elementTypeOid, buffer);
+        break;
+      case (false, true) || (true, false):
+        final value = range.lower ?? range.upper!;
+        _encodeRangeValue(value, encoding, elementTypeOid, buffer);
+      case (true, true):
```

Comment:
let's add explicit `break` here and before the `case` too.

Reply:
done